src/parsers/manifest/dash/__tests__/index.test.ts

```diff
@@ -36,4 +37,89 @@ describe("dash parser", function() {
       }).to.throw("document root should be MPD");
     });
   });
+
+  describe("flattenOverlappingPeriods", function() {
+    // [ Period 1 ][ Period 2 ]       ------>  [ Period 1 ][ Period 3 ]
+    //             [ Period 3 ]
+    it("flatten periods - case 1", function() {
+      const periods = [
+        { id: "1", start: 0, duration: 60, adaptations: {} },
+        { id: "2", start: 60, duration: 60, adaptations: {} },
+        { id: "3", start: 60, duration: 60, adaptations: {} },
+      ];
+
+      const flattenPeriods = flattenOverlappingPeriods(periods);
+      expect(flattenPeriods.length).to.be.equal(2);
+      expect(flattenPeriods[0].start).to.be.equal(0);
+      expect(flattenPeriods[0].duration).to.be.equal(60);
+      expect(flattenPeriods[0].id).to.be.equal("1");
+      expect(flattenPeriods[1].start).to.be.equal(60);
+      expect(flattenPeriods[1].duration).to.be.equal(60);
+      expect(flattenPeriods[1].id).to.be.equal("3");
+    });
+
+    // [ Period 1 ][ Period 2 ]       ------>  [ Period 1 ][  2  ][ Period 3 ]
+    //                  [ Period 3 ]
+    it("flatten periods - case 2", function() {
+      const periods = [
+        { id: "1", start: 0, duration: 60, adaptations: {} },
+        { id: "2", start: 60, duration: 60, adaptations: {} },
+        { id: "3", start: 90, duration: 60, adaptations: {} },
+      ];
+
+      const flattenPeriods = flattenOverlappingPeriods(periods);
+      expect(flattenPeriods.length).to.be.equal(3);
+      expect(flattenPeriods[0].start).to.be.equal(0);
+      expect(flattenPeriods[0].duration).to.be.equal(60);
+      expect(flattenPeriods[0].id).to.be.equal("1");
+      expect(flattenPeriods[1].start).to.be.equal(60);
+      expect(flattenPeriods[1].duration).to.be.equal(30);
+      expect(flattenPeriods[1].id).to.be.equal("2");
+      expect(flattenPeriods[2].start).to.be.equal(90);
+      expect(flattenPeriods[2].duration).to.be.equal(60);
+      expect(flattenPeriods[2].id).to.be.equal("3");
+    });
+
+    // [ Period 1 ][ Period 2 ]       ------>  [  1  ][      Period 3     ]
+    //        [      Period 3     ]
+    it("flatten periods - case 3", function() {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 44. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=394&issues=AWec_EURXtJ12O8uMzxa&open=AWec_EURXtJ12O8uMzxa)

[ISSUE_KEY]: AWec_EURXtJ12O8uMzxa